src/dma.c

```diff
@@ -217,27 +267,39 @@ static uint8_t __io_read8(unsigned long pi_address) {
  * ROM at loading time, a better option is to use DragonFS, where #dfs_read
  * falls back to a CPU memory copy to realign the data when required.
  * 
- * @param[out] ram_address
- *             Pointer to a buffer to place read data
+ * @param[out] ram_pointer
+ *             Pointer to a buffer in RDRAM to place read data
  * @param[in]  pi_address
  *             Memory address of the peripheral to read from
  * @param[in]  len
- *             Length in bytes to read into ram_address
+ *             Length in bytes to read into ram_pointer
  */
-void dma_read_async(void *ram_address, unsigned long pi_address, unsigned long len)
+void dma_read_async(void *ram_pointer, unsigned long pi_address, unsigned long len)
 {
-    unsigned long ram = (unsigned long)UncachedAddr(ram_address);
-    unsigned long rom = pi_address;
+    void *ram = UncachedAddr(ram_pointer);
+    uint32_t ram_address = (uint32_t)ram;
 
     assert(len > 0);
-    assert(((ram ^ rom) & 1) == 0); 
+    assert(((ram_address ^ pi_address) & 1) == 0); 
 
     disable_interrupts();
+
+    // Check if the PI address can be accessed with CPU.
+    // If so, we cannot perform a misaligned transfer.
```

Comment:
shouldn't be the other way around? `if not, we cannot perform...`

Reply:
Fixed.